pkg/drivers/vfkit/vfkit.go

```diff
@@ -313,8 +341,10 @@ func isBootpdError(err error) bool {
 }
 
 func (d *Driver) Stop() error {
-	if err := d.SetVFKitState("HardStop"); err != nil {
-		return err
+	if err := d.SetVFKitState("Stop"); err != nil {
+		// vfkit may be already stopped, shutting down, or not listening.
+		log.Debugf("Failed to set vfkit state to 'Stop': %s", err)
+		return signalPidfile(d.pidfilePath(), syscall.SIGTERM)
 	}
```

Comment:
maybe remove the pid file here so that `GetState` returns `Stopped` after `Stop` was called?

Reply:
signalPidfile removes the pid when the process does not exist. But if the signal is sent the process still exists and the caller will discover it by polling on GetState(). Stop() is not responsible for waiting until the process is stopped, the caller is implementing this for all drivers.

Reply:
What I mean is that if ` d.SetVFKitState("Stop")` succeeds, you have a small window for a race when `GetState` will return `running` while vfkit is `stopping`. It's probably not really important, as `GetState` will eventually get to a consistent state.